code/game/objects/structures/morgue.dm

```diff
@@ -199,29 +199,39 @@ GLOBAL_LIST_EMPTY(crematoriums)
 		locked = 1
 		update_icon()
 
-		for(var/mob/living/M in contents)
+		if (!isnull(contents) && contents.len > 0)
+			cremateLoop(contents)
+
+		new /obj/effect/decal/cleanable/ash(src)
+		sleep(30)
+		if(!QDELETED(src))
+			locked = 0
+			update_icon()
+			playsound(src.loc, 'sound/machines/ding.ogg', 50, 1) //you horrible people
+
+// A recursive proc is needed for checking contents of contents, else bodies won't creamte in body bags
+/obj/structure/bodycontainer/crematorium/proc/cremateLoop(list/conts)
+	for(var/atom/A in conts)
+		if (!isnull(A.contents) && A.contents.len > 0) // Those darn body bags with bodies in them!
+			cremateLoop(A.contents)
+		if (istype(A, /mob/living))
+			var/mob/living/M = A
 			if (M.stat != DEAD)
 				M.emote("scream")
-			if(user)
-				user.log_message("Cremated <b>[M]/[M.ckey]</b>", INDIVIDUAL_ATTACK_LOG)
-				log_attack("\[[time_stamp()]\] <b>[user]/[user.ckey]</b> cremated <b>[M]/[M.ckey]</b>")
+			if(usr)

```

Comment:
we try to avoid the use of the usr global in anything but the procs that absolutely require it (topic usually) as it's a bit of an antipattern.

instead we try to pass through the user as a variable to each proc wherever possible, one because it' makes it possible to pass through other mobs to the proc in other code and two so it's always explicit to follow the chain of where the user variable comes from